### DISCUSSION TITLE
Best practice questions -- nav bars, footers, etc..

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#36775] [Help] [answered]
In an existing React-app I have folder with the name "components" in which I filed items such as nav bars, footers, tables, etc....   usually things that I would re-use in different pages. 

What is the best practice for Next.js ?    Could I do just the same?  would there be any difference in the filing system for these type of items?

**Comment by icyJoseph** [comment, marked as answer]
You can keep the same. How you structure those is not really a concern for Next.js. All that matters is that these are used inside a `pages` to give structure to each route.

Any paradigm you'd like to take works fine. Here's a nice write up by a member of the Vercel team, https://github.com/vercel/swr/discussions/1408#discussioncomment-1383367, though it is in regard to using `swr` in combination with `Next.js`.

Perhaps some things to watch out for are circular imports, and global styles, you'd need to make those available at page level or in `_app` for the entire application.
